### DISCUSSION TITLE
Utilizing trust proxy with secure cookies on serverless

### DISCUSSION THREAD

**Discussion by rossfishkind** [vercel/vercel#3993] [General] [answered]
I'm working on an application that uses Next.js and Now to deploy a serverless API. The issue I seem to have found is that when authenticating with the API using a `secure` cookie, it doesn't seem to respect the session. I believe this is because you typically need to utilize the `trust proxy` setting on the server in order to tell the cookie that it's ok to send over the secure connection.

I was wondering if anyone has gotten around this issue, or if there is a way to set the `trust proxy` setting when utilizing serverless lambdas.

Thanks!

**Comment by paulogdm** [comment, marked as answer]
The `now.sh` links are in the public suffix list as well. Maybe that can affect your current implementation? A [thread](https://twitter.com/wesbos/status/1062743502939922432) made a long time ago about this.

**Reply by rossfishkind** [reply]
Ah yes, I think that may be it. Thanks @paulogdm!
